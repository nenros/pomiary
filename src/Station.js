import React from "react";
import OptionList from './OptionList'
import ActivityIcon from './ActivityIcon'
import ExpandedData from './ExpandedData.js'

const Station = ({ station }) => {
    const { name, active, no } = station
    const options = ['rain', 'water', 'winddir', 'windlevel']
    return <div className="column is-half">
        <div className="box">
            <div className="content">
                <div>
                    <strong>{name}</strong>
                    <ActivityIcon active={active} />
                    <br />
                    Opcje:
                    <OptionList object={station} options={options} />
                    {options.map((option) => <ExpandedData key={`${no}_${option}`} id={no} option={option} />)}
                </div>
            </div>
        </div>
    </div>
}

export default Station
