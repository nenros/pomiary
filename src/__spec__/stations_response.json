{
    "success": {
        "status": "success",
        "message": "Lista stacji",
        "data": [
            {
                "active": true,
                "name": "GÃ³rki Zachodnie",
                "no": 2,
                "rain": false,
                "water": false,
                "winddir": false,
                "windlevel": false
            }
        ]
    },
    "error": {
        "status": "error",
        "message": "cos poszlo nie tak"
    }
}
