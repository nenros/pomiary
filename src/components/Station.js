import React from "react";
import OptionList from './OptionList'
import ActivityIcon from './ActivityIcon'
import ExpandedData from './ExpandedData.js'
import Card from './hoc/Card'
import Tabs from './hoc/Tabs'

const Station = ({ station }, { measureNameToOption }) => {
    const { name, active, no } = station
    const options = ['rain', 'water', 'winddir', 'windlevel']
    return <div className="column is-half">
        <Card>
            <p>
                <strong>{name}</strong>
                <ActivityIcon active={active} />
            </p>
            <section>
                <Tabs>
                    <span>Opcje</span>
                    <OptionList object={station} options={options} />
                    {options.map((option) =>
                        <React.Fragment key={`${no}_${option}`}>
                            <span>{option}</span>
                            <ExpandedData id={no} option={option} />
                        </React.Fragment>)}
                </Tabs>
            </section>
        </Card>
    </div>
}

export default Station
