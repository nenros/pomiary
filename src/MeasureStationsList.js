import React, { Component } from 'react'
import Station from "./Station";

class MeasureStationsList extends Component {
    static API_ENDPOINT = '/stations'

    state = {
        stations: []
    }

    async componentDidMount() {
        await this.fetchStations()
    }

    async fetchStations() {
        try {
            const response = await fetch(MeasureStationsList.API_ENDPOINT)
            const body = await response.json()
            if (body.status !== "success") {
                throw new Error(body.message)
            }
            return this.setState({ ...this.state, stations: body.data })

        } catch (error) {
            console.log(error)
        }
    }

    render() {
        const { stations } = this.state
        return <div className="columns is-multiline">{
            stations.map((station) => <Station key={station.no} station={station} />)
        }</div>
    }
}

export default MeasureStationsList
