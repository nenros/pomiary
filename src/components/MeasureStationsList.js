import React, { Component } from 'react'
import Station from "./Station";
import { fetchStations } from '../actions/stations'
import { connect } from 'react-redux'


export class MeasureStationsList extends Component {

    componentDidMount() {
        this.props.fetchStations()
    }

    render() {
        const { records, error } = this.props.stations
        if (error) {
            return <section className="message is-danger">
                <div className="message-header">
                    <p>Error</p>
                </div>
                <div className="message-body">
                    {error}
                </div>
            </section>
        }
        return <div className="columns is-multiline">{
            records.map((station) => <Station key={station.no} station={station} />)
        }</div>
    }
}

export default connect(
    state => ({ stations: state.stations }),
    (dispatch) => ({ fetchStations: () => dispatch(fetchStations()) })
)(MeasureStationsList)

