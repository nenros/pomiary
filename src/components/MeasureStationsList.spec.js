import React from 'react';
import { shallow } from 'enzyme';
import { MeasureStationsList } from './MeasureStationsList'
import Station from './Station'

import * as stationsResponse from '../__spec__/stations_response.json'

const defaultProps = {
    fetchStations: jest.fn(),
    stations: {
        error: null,
        records: stationsResponse.success.data
    }
}


const setup = (props = defaultProps) => {
    return shallow(<MeasureStationsList {...props} />)
}

describe("MeasureStationsList", () => {
    it("should render list", () => {
        const enzymeWrapper = setup()
        expect(enzymeWrapper.find("Station")).toHaveLength(1)
        expect(enzymeWrapper).toMatchSnapshot()
    })

    it("should render station", () => {
        const enzymeWrapper = setup()
        expect(enzymeWrapper).toContainReact(<Station station={defaultProps.stations.records[0]} />)
    })

    it("should render station", () => {
        const error = "error message"
        const props = { ...defaultProps, stations: { ...defaultProps.stations, error } }
        const enzymeWrapper = setup(props)
        expect(enzymeWrapper.find('.message-body')).toHaveText(error)
    })
})
