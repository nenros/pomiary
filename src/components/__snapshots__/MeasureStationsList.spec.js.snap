// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasureStationsList should render list 1`] = `
<div
  className="columns is-multiline"
>
  <Station
    key="2"
    station={
      Object {
        "active": true,
        "name": "GÃ³rki Zachodnie",
        "no": 2,
        "rain": false,
        "water": false,
        "winddir": false,
        "windlevel": false,
      }
    }
  />
</div>
`;
